catalog/app/containers/Admin/Buckets.tsx

```diff
@@ -367,21 +455,128 @@ function BucketFields({ name, reindex }: BucketFieldsProps) {
         </M.AccordionSummary>
         <M.Box className={classes.group} pt={1}>
           {!!reindex && (
-            <M.Button variant="outlined" fullWidth onClick={reindex}>
-              Re-index and repair
-            </M.Button>
+            <M.Box pb={2.5}>
+              <M.Button variant="outlined" fullWidth onClick={reindex}>
+                Re-index and repair
+              </M.Button>
+            </M.Box>
           )}
+
           <RF.Field
-            component={Form.Field}
-            name="fileExtensionsToIndex"
-            label="File extensions to index (comma-separated)"
-            fullWidth
-            margin="normal"
+            component={Form.Checkbox}
+            type="checkbox"
+            name="enableDeepIndexing"
+            label={
+              <>
+                Enable deep indexing
+                <Hint>Deep indexing help text TBD @akarve pls halp!</Hint>
+              </>
+            }
           />
+
+          <RF.FormSpy subscription={{ modified: true, values: true }}>
+            {({ modified, values }) => {
+              // don't need this while adding a bucket
+              if (!bucket) return null
+              if (
+                !modified?.enableDeepIndexing &&
+                !modified?.fileExtensionsToIndex &&
+                !modified?.indexContentBytes
+              )
+                return null
+              try {
+                if (
+                  R.equals(
+                    bucket.fileExtensionsToIndex,
+                    editFormSpec.fileExtensionsToIndex(values),
+                  ) &&
+                  R.equals(
+                    bucket.indexContentBytes,
+                    editFormSpec.indexContentBytes(values),
+                  )
+                )
+                  return null
+              } catch {
+                return null
```

Comment:
Maybe, at least, make `console.error`?

Reply:
when it throws, it means the input is invalid (cannot be converted to a proper output format), so it's just a validation error -- we don't want to log validation errors